UefiCpuPkg/PiSmmCpuDxeSmm/CpuS3.c

```diff
@@ -824,9 +827,25 @@ SmmRestoreCpu (
   }
 
   //
-  // Restore SMBASE for BSP and all APs
+  // Retrive the allocated SmmBase from gSmmRelocationInfoHobGuid
+  //
+  GuidHob = GetFirstGuidHob (&gSmmRelocationInfoHobGuid);
+  if (GuidHob != NULL) {
+    mSmmRelocated = TRUE;
+  } else {
+    mSmmRelocated = FALSE;
+  }
+
   //
-  SmmRelocateBases ();
+  // Check whether Smm Relocation is done or not.
+  // If not, will do the SmmBases Relocation here!!!
+  //
+  if (!mSmmRelocated) {
+    //
+    // Restore SMBASE for BSP and all APs
+    //
+    SmmRelocateBases ();
+  }

```

Comment:
I notice in PiCpuSmmEntry, we only skipped part of the SmmRelocateBases, remaining work is done in SmbaseRelocationPostSmmInit(). Should we do the same in s3 path?

Reply:
yes, thanks catch, i will handle the logic in new patch.